From 8d35f4dfb79b2b21e4979a2188a037bdf667c080 Mon Sep 17 00:00:00 2001
From: Thiraviyam Mariyappan <quic_tmariyap@quicinc.com>
Date: Tue, 28 Jan 2025 15:40:18 +0530
Subject: UPSTREAM:DP:wifi:mac80211: Disable LLTX feature

Use of LLTX (LockLess Transmit) is to improve the performance of
network device drivers by avoiding the use of locks during packet
transmission. This feature is deprecated now. Hence, Disable this
feature.

Signed-off-by: Thiraviyam Mariyappan <quic_tmariyap@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/mac.c | 2 --
 net/mac80211/ieee80211_i.h            | 3 +--
 2 files changed, 1 insertion(+), 4 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/mac.c b/drivers/net/wireless/ath/ath12k/mac.c
index 20add83..c061a46 100644
--- a/drivers/net/wireless/ath/ath12k/mac.c
+++ b/drivers/net/wireless/ath/ath12k/mac.c
@@ -21122,8 +21122,6 @@ static int ath12k_mac_hw_register(struct ath12k_hw *ah)
 		ieee80211_hw_set(hw, SUPPORT_FAST_XMIT);
 	}
 
-	hw->netdev_features |= NETIF_F_LLTX;
-
 	ieee80211_hw_set(hw, SUPPORT_ECM_REGISTRATION);
 
 	ret = ieee80211_register_hw(hw);
diff --git a/net/mac80211/ieee80211_i.h b/net/mac80211/ieee80211_i.h
index 5ddb88a..0855d73 100644
--- a/net/mac80211/ieee80211_i.h
+++ b/net/mac80211/ieee80211_i.h
@@ -2096,8 +2096,7 @@ void ieee80211_color_collision_detection_work(struct work_struct *work);
 /* interface handling */
 #define MAC80211_SUPPORTED_FEATURES_TX	(NETIF_F_IP_CSUM | NETIF_F_IPV6_CSUM | \
 					 NETIF_F_HW_CSUM | NETIF_F_SG | \
-					 NETIF_F_HIGHDMA | NETIF_F_GSO_SOFTWARE | \
-					 NETIF_F_LLTX)
+					 NETIF_F_HIGHDMA | NETIF_F_GSO_SOFTWARE)
 #define MAC80211_SUPPORTED_FEATURES_RX	(NETIF_F_RXCSUM)
 #define MAC80211_SUPPORTED_FEATURES	(MAC80211_SUPPORTED_FEATURES_TX | \
 					 MAC80211_SUPPORTED_FEATURES_RX)
-- 
2.34.1

